File: src/main/java/io/github/docs/app/model/FormalDocumentMetadata.java
package io.github.docs.app.model;

import io.github.docs.domain.enumeration.DocumentType;
import io.github.docs.service.dto.SchemeDTO;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Lob;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;

/**
 * Metadata about a formal document
 */
@Data
@NoArgsConstructor
public class FormalDocumentMetadata {

    private Long id;

    @NotNull
    private String documentTitle;

    private String documentSubject;

    private String briefDescription;

    private LocalDate documentDate;

    private DocumentType documentType;

    private String documentStandardNumber;

    @NotNull
    private String filename;

    private Set<SchemeDTO> schemes = new HashSet<>();

    @Override
    public String toString() {
        final StringBuilder sb = new StringBuilder("FormalDocumentMetadata{");
        sb.append("id=").append(id);
        sb.append(", documentTitle='").append(documentTitle).append('\'');
        sb.append(", documentSubject='").append(documentSubject).append('\'');
        sb.append(", briefDescription='").append(briefDescription).append('\'');
        sb.append('}');
        return sb.toString();
    }
}

File: src/main/webapp/app/bespoke/model/formal-document-metadata.ts
import { Moment } from 'moment';
import { IScheme } from 'app/shared/model/scheme.model';
import { DocumentType } from 'app/shared/model/enumerations/document-type.model';

export interface IFormalDocumentMetadata {
  id?: number;
  documentTitle?: string;
  documentSubject?: string;
  briefDescription?: string;
  documentDate?: Moment;
  documentType?: DocumentType;
  documentStandardNumber?: string;
  filename?: string;
  schemes?: IScheme[];
}

export class FormalDocumentMetadata implements IFormalDocumentMetadata {
  constructor(
    public id?: number,
    public documentTitle?: string,
    public documentSubject?: string,
    public briefDescription?: string,
    public documentDate?: Moment,
    public documentType?: DocumentType,
    public documentStandardNumber?: string,
    public filename?: string,
    public schemes?: IScheme[]
  ) {}
}
